falcon/util/uri.py

```diff
@@ -26,6 +26,10 @@
 
 import six
 
+
+# NOTE(kandziu): Invalid unicode character
+FFFD = u'\ufffd'
```

Comment:
Any reason for singling out ``FFFD``?
There are more non-characters in Unicode: https://en.wikipedia.org/wiki/Universal_Character_Set_characters#Non-characters

What is more, do these pose any practical problem for URI handling? We could make sure we do not allow them inside routes though.

Reply:
By default unicode encoder replaces invalid characters on `U+FFFD` [refs](https://docs.python.org/2/howto/unicode.html#the-unicode-type).  How we should process a such `/static/.\ufffdssh/authorized_keys` uri path?

Reply:
I am no expert on this tbh. But have you seen [Corrigendum #9: Clarification About Noncharacters](https://www.unicode.org/versions/corrigendum9.html) though? They are not invalid _per se_, just private.

I have to apologize though I haven't notice in the beginning these you apply it only to static [file] routes. It makes sense in this case I guess.

If the path is now strictly guaranteed to be ``unicode`` though, could we just compile this ``U+FFFD`` into ``StaticRoute._DISALLOWED_CHARS_PATTERN``?

Reply:
Yes we can compile it into `StaticRoute._DISALLOWED_CHARS_PATTERN`